rpc/lib/server/http_server.go

```diff
@@ -48,7 +72,7 @@ func StartHTTPServer(listener net.Listener, handler http.Handler, logger log.Log
 	s := &http.Server{
```

Comment:
how about we refactor StartHTTPServer to accept config struct

```go
type config struct {
  WriteTimeout int
  ReadTimeout int
}
```

and return http.Server, which we'll use in RPC to get Timeout value. I don't really like global variables

Reply:
I thought about this too. But this would come at the cost of more refactoring: 

`StartHTTPServer` in its current form blocks (does no return) as it calls `http.Server.Serve`. 
Sure, we could change that. Then, we would need to rename the method to `NewHTTPServer` as it does not start anything anymore and update all places to manually call `Serve`. Same holds for `StartHTTPAndTLSServer`. 

> which we'll use in RPC to get Timeout value

This assumes the places were we need to read the timeout value have access to the returned `http.Server` (meaning they call `NewHTTPServer`). Also, not the case currently. 

I'd vouch to keep this in mind and open an issue about a larger refactoring here. It's also likely, there are more places in the rpc package like this (there are quite a few gloabal vars).

Reply:
> StartHTTPServer in its current form blocks 

oh, right. forgot about that.

but we can still pass a config struct

Reply:
We could indeed but what would we gain from that? Currently the values are read from a global var and directly shared with other places via that global vars (I agree, this is very bad). This PR makes one of these global vars unexported s.t. that the package is "in control". Having a config as you described makes sense but how do we then ensure that increased the writeTimeout if necessary (`WriteTimeout > TimeoutBroadcastTxCommit`)? Using the adjusted value from the global var in the passed config? It gets even more confusing IMHO.

Reply:
https://github.com/tendermint/tendermint/pull/3443/commits/811d92c1bd4d32b37162ee5eccab6348f9ba5e5d#r266826680